core/templates/dev/head/profile/Preferences.js

```diff
@@ -35,6 +35,10 @@ oppia.controller('Preferences', ['$scope', '$http', '$rootScope', '$modal', '$ti
     _saveDataItem('user_bio', userBio);
   };
 
+  $scope.saveSubjectInterests = function(subjectInterests) {
```

Comment:
I'm not sure how this works. I went to test the username, and did something purposely erroneous, and yet no warning came up, as I assume the filter should do.